pkg/cloudprovider/aws/packing/packing.go

```diff
@@ -78,6 +80,42 @@ func (p *podPacker) Pack(ctx context.Context, pods []*v1.Pod) ([]*Packing, error
 
 // TODO filter instance types based on node contraints like availability zones etc.
 func (p *podPacker) getNodeCapacities() []*nodeCapacity {
+	nodeCapacities := []*nodeCapacity{}
+	describeInstanceTypesInput := &ec2.DescribeInstanceTypesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("processor-info.supported-architecture"),
+				Values: []*string{aws.String("x86_64")},
+			},
+			{
+				Name:   aws.String("supported-usage-class"),
+				Values: []*string{aws.String("on-demand")},
+			},
+			{
+				Name:   aws.String("supported-virtualization-type"),
+				Values: []*string{aws.String("hvm")},
+			},
+		},
+	}
+	err := p.ec2.DescribeInstanceTypesPagesWithContext(context.TODO(), describeInstanceTypesInput, func(page *ec2.DescribeInstanceTypesOutput, lastPage bool) bool {
+		for _, instanceTypeInfo := range page.InstanceTypes {
+			nodeCapacity, err := instanceTypeInfoToNodeCapacity(*instanceTypeInfo)
+			if err != nil {
+				zap.S().Warnf("Failed to convert instanceTypeInfo to a nodeCapacity, %s", err.Error())
```

Comment:
Maybe controversial, but I think we should fail fast and hard here.

Reply:
That's a fair point, I can change it. By failing fast and hard I assume you just mean to propagate an error up?

Reply:
Depends when this runs. I see two approaches
1) Call only on startup and panic to fail fast
2) Call occasionally, with caching, similar subnets/launchtemplates, return an error in the reconcile loop on error

Reply:
I think #2 would be more useful in practice - there's no reason this API should fail on any continual basis.

Reply:
👍  This is definitely very cachable data. The data shouldn't change at a high velocity.   

Some reasons why this would fail would be rate limiting, which would obviously be temporary, and permissions issues.  I've seen that some paths aren't exercised regularly, so it may be nice to also call on startup all APIs we're using with dry-runs to ensure permissions are good on startup.

Reply:
I think it's better - if there are permissions issues - to just very clearly fail with an error the operator can see and fix, I don't really see how the controller can proceed without this info (for example it doesn't really know what instance types are available in a given region unless it can call this).

Reply:
ok, I'll make that change 👍

Reply:
Also -- in the spirit of reconciliation, if permissions are missing, the controller should eventually recover if they're added -- it shouldn't require a restart or any manual intervention.